### PR TITLE
loading progress

### PR DESCRIPTION
**Pull request by dek3rr**
Here's a working version of #768 
It's not everything I had envisioned, but unfortunately, my ubuntu hard drive died shortly after. 
I still think it's useful in it's current state, so I'm putting up the PR sans progress bar for now.

### CODE REVIEW

**Review by pd0wm — CHANGES_REQUESTED** [review]

File: selfdrive/manager.py [hunk 1]
```diff
@@ -504,8 +504,22 @@ def manager_prepare():
...
+  
+  params = Params()
+  process_cnt = len(managed_processes)
+  loader_proc = subprocess.Popen(["./spinner"], stdin=subprocess.PIPE,
+        cwd=os.path.join(BASEDIR, "selfdrive", "ui", "spinner"),
+        close_fds=True)
+  spinner_text = "chffrplus" if params.get("Passive")=="1" else "openpilot"
+  
+  for n,p in enumerate(managed_processes):
+    if os.getenv("PREPAREONLY") is None:
```
[the remark this answers is not in the document]
Comment (pd0wm) [reply selfdrive/manager.py]: You're right!

File: selfdrive/ui/spinner/spinner.c [hunk 2]
```diff
@@ -16,13 +16,14 @@
 
 #include "common/framebuffer.h"
 
-int main(int argc, char** argv) {
+int main() {
   int err;
-
-  const char* spintext = NULL;
-  if (argc >= 2) {
```
[the remark this answers is not in the document]
Comment (pd0wm) [reply selfdrive/ui/spinner/spinner.c]: That would be nice. That keeps the spinner versatile and easy to call from a shells script.

File: selfdrive/manager.py [hunk 3]
```diff
@@ -504,8 +504,22 @@ def manager_prepare():
 
   # build all processes
   os.chdir(os.path.dirname(os.path.abspath(__file__)))
-  for p in managed_processes:
+  
+  params = Params()
+  process_cnt = len(managed_processes)
+  loader_proc = subprocess.Popen(["./spinner"], stdin=subprocess.PIPE,
```
[the remark this answers is not in the document]
Comment (pd0wm) [reply selfdrive/manager.py]: Didn't mean you should use tqdm, just an example of the API you could use.

**Comment by ErichMoraga** [comment]
Forget about the spinner.  Text detailing what processes are loading, etc. would be sufficient.

**Comment by pd0wm** [comment]
The spinner was refactored a little bit, so we can also use it in the NEOS installer. Unfortunately we forgot to open source the files after they were moved. I just pushed those files to devel (#812).

**Comment by pd0wm** [comment]
I'll finish the PR and merge it internally! I'll probably add a progress bar too.

**Comment by pd0wm** [comment]
Thanks @dekerr for figuring out the stdin stuff!

ezgif-2-35311e685540

**Comment by dekerr** [comment]
Awesome. Thanks, Willem. Looks great.

**Pull request closed without merging**